core/src/main/java/org/opensearch/sql/calcite/utils/BuiltinFunctionUtils.java

```diff
@@ -99,6 +118,30 @@ static SqlOperator translate(String op) {
   static List<RexNode> translateArgument(
       String op, List<RexNode> argList, CalcitePlanContext context) {
     switch (op.toUpperCase(Locale.ROOT)) {
+      case "TRIM":
+        List<RexNode> trimArgs =
+            new ArrayList<>(
+                List.of(
+                    context.rexBuilder.makeFlag(SqlTrimFunction.Flag.BOTH),
+                    context.rexBuilder.makeLiteral(" ")));
+        trimArgs.addAll(argList);
+        return trimArgs;
+      case "LTRIM":
+        List<RexNode> LTrimArgs =
+            new ArrayList<>(
+                List.of(
+                    context.rexBuilder.makeFlag(SqlTrimFunction.Flag.LEADING),
+                    context.rexBuilder.makeLiteral(" ")));
+        LTrimArgs.addAll(argList);
+        return LTrimArgs;
+      case "RTRIM":
+        List<RexNode> RTrimArgs =
+            new ArrayList<>(
+                List.of(
+                    context.rexBuilder.makeFlag(SqlTrimFunction.Flag.TRAILING),
+                    context.rexBuilder.makeLiteral(" ")));
+        RTrimArgs.addAll(argList);
+        return RTrimArgs;
```

Comment:
nit, [Calcite defined RTRIM and LTRIM](https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java#L193) , could we consider reuse it in future?

Reply:
In our current implementation, we will use the `RexCallImplementor` in https://github.com/apache/calcite/blob/48648b391367eeb7910e934cc73dabbe1d7370c0/core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java#L1416, and the ltrim and rtrim is not actually defined in the `RexImpTable`. So these two functions are not suitable for our current implementation. But we could track them and see whether we could change our logic to reuse them.